### PR TITLE
add Image::bumpmap_to_normalmap conversion function

### PR DESCRIPTION
**Pull request by Lerc**

### CODE REVIEW

**Comment by akien-mga** [comment]
Thanks for your contribution! There are various style issues ([diff](https://travis-ci.org/godotengine/godot/jobs/344504072)), you should use `clang-format` 5.0 to format your code (see [code style guidelines](http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html)).

File: core/image.cpp [hunk 1]
```diff
@@ -2379,6 +2380,47 @@ void Image::normalmap_to_xy() {
...
+				float here = read_ptr[ty * width + tx];
+				float to_right = read_ptr[ty * width + px];
+				float above = read_ptr[py * width + tx];
+				Vector3 up = Vector3(0, 1, (here - above) * bump_scale);
+				Vector3 across = Vector3(1, 0, (to_right - here) * bump_scale);
+
+				Vector3 normal = across.cross(up);
+				normal.normalize();
+
+				write_ptr[((ty * width + tx) << 2) + 0] = (128.0 + normal.x * 127.0);
```
Comment (tagcup) [inline core/image.cpp]: This maps [-1,1] to [1,255] rather than [0,255]

Comment (Lerc) [reply core/image.cpp]: Good catch.   (127.5 + normal.x * 127.5)  is right, yes?

Comment (tagcup) [reply core/image.cpp]: Looks right to me, up to possible rounding issues

**Pull request merged**